### PR TITLE
Fix for #9

### PR DESCRIPTION
**Pull request by nhz-io**

### CODE REVIEW

**Comment by nhz-io** [comment]
To minimize the impact on the rest of the code, i kept the aliases as they are (non-flat, nested).
So the fix is only on the `flags.string`. The fix is future proof if one day aliases become flat.

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
 		"prepublish": "not-in-publish || npm run prepublishOnly",
 		"lint": "eslint --ext=js,mjs .",
 		"pretest": "npm run lint",
-		"tests-only": "nyc tape test/*.js",
+		"tests-only": "nyc tape test/*.js test/**/*.js",
```
Comment (ljharb) [inline package.json]:
```suggestion
		"tests-only": "nyc tape 'test/**/*.js'",
```

File: test/fixes/issue-09.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var test = require('tape');
```
Comment (ljharb) [inline test/fixes/issue-09.js]: i don't think we need a special folder just for "fixes"; this should just be part of normal parsing tests (`test/parse.js`)

**Comment by shadowspawn** [comment]
I prefer expanding the tests to cover this combination rather than adding a specific regression test for the failure. So the test would be named like "opt.string works with multiple aliases" rather than "Issue #9 fixed".

But wait for a second opinion from @ljharb before changing.

File: test/fixes/issue-09.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+var parse = require('../..');
+
+test('Issue #9 fixed', function (t) {
```
Comment (ljharb) [inline test/fixes/issue-09.js]: the test name should describe what it's testing; "which issue it is" can be found via the git log :-)

**Review by ljharb — CHANGES_REQUESTED** [review]
Fix looks great and backwards-compatible!

File: index.js [hunk 4]
```diff
@@ -61,7 +61,9 @@ module.exports = function (args, opts) {
 	[].concat(opts.string).filter(Boolean).forEach(function (key) {
 		flags.strings[key] = true;
 		if (aliases[key]) {
-			flags.strings[aliases[key]] = true;
+			[].concat(aliases[key]).forEach(function (k) {
```
Comment (shadowspawn) [inline index.js:64]: I like this as being consistent with the existing code, and localised.

File: test/fixes/issue-09.js (hunk 2 above)
Comment (nhz-io) [reply test/fixes/issue-09.js]: done

File: test/fixes/issue-09.js (hunk 3 above)
Comment (nhz-io) [reply test/fixes/issue-09.js]: its under the **string and alias** group now

File: test/parse.js [hunk 5]
```diff
@@ -142,6 +142,18 @@ test('string and alias', function (t) {
...
 	t.equal(y.s, '000123');
 	t.equal(typeof y.s, 'string');
+
+	var z = parse(['-s123'], {
+		alias: { str: ['s', 'S'] },
+		string: ['str'],
+	});
+
+	t.equal(z.str, '123');
+	t.equal(typeof z.str, 'string');
```
Comment (ljharb) [inline test/parse.js]: equal is strict, so we don’t need to check the type separately.

we’d also get a better error message if we used deepEqual on the entire z object.

**Review by ljharb — APPROVED** [review]

File: test/parse.js (hunk 5 above)
Comment (nhz-io) [reply test/parse.js]: agree

**Review by ljharb — APPROVED** [review]
LGTM!

**Pull request merged**